cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/Database200MigrationTest.java

```diff
@@ -0,0 +1,530 @@
+/**
+ * Copyright (C) 2016 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore;
+
+import static org.hamcrest.Matchers.is;
+
+import com.cloudant.mazha.Document;
+import com.cloudant.sync.datastore.migrations.MigrateDatabase100To200;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.sqlite.SQLDatabaseQueue;
+import com.cloudant.sync.sqlite.SQLQueueCallable;
+import com.cloudant.sync.util.DatabaseUtils;
+import com.cloudant.sync.util.TestUtils;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Created by rhys on 15/08/2016.
+ */
+public class Database200MigrationTest {
+
+    private Datastore ds;
+    private DatastoreManager manager;
+    private DocumentRevision rootRevision;
+    private String dir;
+
+    @Before
+    public void setUp() throws Exception {
+        dir = TestUtils.createTempTestingDir(this.getClass().getName());
+        manager =  DatastoreManager.getInstance(dir);
+        ds = manager.openDatastore("complexTree");
+
+        Map<String,String> body = new HashMap<String,String>();
+        body.put("hello","world");
+
+        DocumentRevision rev = new DocumentRevision();
+        rev.setBody(DocumentBodyFactory.create(body));
+        rootRevision = ds.createDocumentFromRevision(rev);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ds.close();
+        TestUtils.deleteTempTestingDir(dir);
+    }
+
+    /**
+     * Tree:
+     *
+     * abc:
+     *
+     * 1-xxxxxxx
+     *
+     * 1-xxxxxxx
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDatabaseWithComplexTree() throws Exception {
+
+        // sets up the datastore with 1-x and 1-x revisions (Duplicates)
+
+        final String sql = "INSERT INTO revs (doc_id, parent, current, deleted, available, revid, json) SELECT doc_id, parent, current, deleted, available, revid, json FROM revs where revs.revid = ?";
```

Comment:
This constant is repeated in many tests, I suggest pulling it up to the class level.